app/src/main/java/com/github/se/wanderpals/model/viewmodel/MapViewModel.kt

```diff
@@ -30,16 +38,82 @@ open class MapViewModel(tripsRepository: TripsRepository, private val tripId: St
   }
 
   init {
+    refreshData()
+  }
+
+  /** Refresh the data. */
+  fun refreshData() {
     getAllStops()
+    getAllSuggestions()
+    getAllUsersPositions()
   }
 
   /** Get all stops from the trip. */
   open fun getAllStops() {
     viewModelScope.launch {
       val allStops = _tripsRepository.getAllStopsFromTrip(tripId)
-      val allStopsFromSubscribedTrips =
-          _tripsRepository.getAllSuggestionsFromTrip(tripId).map { it.stop }
-      stops.value = allStops.toMutableList().apply { addAll(allStopsFromSubscribedTrips) }
+      stops.value = allStops
```

Comment:
Stops could also have a (0,0) coordinates (which is our invalid case), in which case they should also not be displayed. You can similarly to your suggestion code, add a map and a filter)